storage_thumbnail/models/thumbnail_mixin.py

```diff
@@ -0,0 +1,132 @@
+        "thumb_medium_id", "thumb_small_id", "backend_id.backend_view_use_internal_url"
+    )
+    def _compute_thumb_urls(self):
+        for backend, records in tools.groupby(self, lambda x: x.backend_id):
+            url_fname = "url"
+            if backend.backend_view_use_internal_url:
+                url_fname = "internal_url"
+            for rec in records:
+                rec.image_medium_url = rec.thumb_medium_id[url_fname]
+                rec.image_small_url = rec.thumb_small_id[url_fname]
+
+    def _get_thumb(self, scale_key=None, scale=None):
+        """Retrievet the first thumb matching given scale."""
+        assert scale_key or scale
+        scale = scale or self._image_scale_mapping[scale_key]
+        size_x, size_y = scale
+        for thumb in self.thumbnail_ids:
+            if thumb.size_x == size_x and thumb.size_y == size_y:
+                return thumb
+
+    def _get_medium_thumbnail(self):
+        return self.get_or_create_thumbnail(*self._image_scale_mapping["medium"])
+
+    def _get_small_thumbnail(self):
+        return self.get_or_create_thumbnail(*self._image_scale_mapping["small"])
+
+    def _get_url_key(self, url_key):
+        if url_key:
+            url_key = slugify(url_key)
+        return url_key
+
+    def get_existing_thumbnail(self, size_x, size_y, url_key=None):
+        self.ensure_one()
+        url_key = self._get_url_key(url_key)
+        thumbnail = self.env["storage.thumbnail"].browse()
+        for th in self.thumbnail_ids:
+            if th.size_x == size_x and th.size_y == size_y:
+                if url_key and url_key != th.url_key:
+                    continue
+                thumbnail = th
+                break
+        return thumbnail
+
+    def get_or_create_thumbnail(self, size_x, size_y, url_key=None):
+        url_key = self._get_url_key(url_key)
+        thumbnail = self.get_existing_thumbnail(size_x, size_y, url_key=url_key)
+        if not thumbnail and self.data:
+            vals = self.env["storage.thumbnail"]._prepare_thumbnail(
+                self, size_x, size_y, url_key
+            )
+            thumbnail = self.thumbnail_ids.create(vals)
+            # invalidate field since a new record is created
+            # The actual model is a mixin, therefore the inverse into
+            # storage.thumbnail is not defined as a one2many to this mixin.
+            # As consequence, the ORM is not able to trigger the invalidation
+            # of thumbnail_ids on our mixin
+            self.invalidate_recordset(fnames=["thumbnail_ids"])
+        return thumbnail
+
+    def generate_odoo_thumbnail(self):
+        self_sudo = self.sudo()
+        self_sudo._get_small_thumbnail()
+        self_sudo._get_medium_thumbnail()
+        return True
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        records = super().create(vals_list)
+
+        for record in records:
+            record.generate_odoo_thumbnail()
+
+        return records
```

Comment:
```suggestion
        records = super().create(vals_list)
        for record in records:
            record.generate_odoo_thumbnail()
        return records
```

nitpicking: extra spaces